src/main/groovy/edu/oregonstate/mist/inventory/db/InventoryDAOWrapper.groovy

```diff
@@ -112,15 +115,112 @@ class InventoryDAOWrapper {
     }
 
     /**
+     * Update a single inventory object by ID
+     * @param inventory
+     * @param inventoryID
+     */
+    @Transaction
+    public void updateInventory(Inventory inventory, String inventoryID) {
+        //Update top level inventory attributes
+        inventory.trimNameAndDescription()
+        inventoryDAO.updateInventory(inventory, inventoryID)
+
+        //Update api query parameters
+        createUpdateDeleteFields(inventory.apiQueryParams, inventoryID, inventoryID, QUERY_DB_TYPE)
+
+        //Update consuming entities
+        List<ConsumingEntity> currentConsumingEntities =
+                inventoryDAO.getConsumingEntities(inventoryID)
+
+        def currentConsumingEntityIDs = currentConsumingEntities.collect { it.entityID }
+
+        inventory.consumingEntities.each { consumingEntity ->
+            if (currentConsumingEntityIDs.contains(consumingEntity.entityID)) {
+                inventoryDAO.updateConsumingEntity((ConsumingEntity) consumingEntity, inventoryID)
+            } else {
+                inventoryDAO.createConsumingEntity((ConsumingEntity) consumingEntity, inventoryID)
+            }
+        }
+
+        def consumingEntityIDsToDelete = currentConsumingEntityIDs -
+                inventory.consumingEntities.collect { it.entityID }
+
+        consumingEntityIDsToDelete.each {
+            inventoryDAO.deleteConsumingEntities(inventoryID, it)
+        }
+
+        //Update provided data
+        List<DataSource> currentProvidedData = inventoryDAO.getProvidedData(inventoryID)
+
+        def currentProvidedDataIds = currentProvidedData.collect { it.sourceID }
+
+        inventory.providedData.each { dataSource ->
+            if (currentProvidedDataIds.contains(dataSource.sourceID)) {
+                inventoryDAO.updateProvidedData((DataSource) dataSource, inventoryID)
+                //Update provided data fields
+                createUpdateDeleteFields(
+                        dataSource.fields, inventoryID, dataSource.sourceID, PROVIDED_DATA_DB_TYPE)
+            } else {
+                inventoryDAO.createProvidedData((DataSource) dataSource, inventoryID)
+
+                dataSource.fields.each { field ->
+                    inventoryDAO.createField(
+                            (Field) field, dataSource.sourceID, PROVIDED_DATA_DB_TYPE, inventoryID)
+                }
+            }
+        }
+
```

Comment:
My recommendation would be to have the different pieces of logic in this method split into separate smaller methods that this one calls: updateProvidedData(), updateConsumingEntities() and updateFields(). That would simplify this method a bit.